pkg/cmd/cli/cmd/newapp.go

```diff
@@ -531,8 +533,20 @@ func handleRunError(c *cobra.Command, err error, fullName string) error {
 	}
 	switch t := err.(type) {
 	case newcmd.ErrRequiresExplicitAccess:
+		if t.Input.Token != nil && t.Input.Token.ServiceAccount {
+			return fmt.Errorf(`installing %q requires an 'installer' service account with project editor access
+
+WARNING: This will allow the pod to create and manage resources within your namespace -
+ensure you trust the image with those permissions before you continue.
+
+You can see more information about the image by adding the --dry-run flag.
+If you trust the provided image, include the flag --grant-install-rights.`, t.Match.Value)
+		}
```

Comment:
@smarterclayton just one more nit about this message. It'd be nice if it told me that if I specify --grant-install-rights, a service account will be created for me if one doesn't already exist.